app/views/layouts/embed.html.erb

```diff
@@ -38,8 +38,12 @@
         }
 
         window.onload = function() {
-          // Send a post message with our loaded height
-          postUp({type: 'discourse-resize', height: document['body'].offsetHeight});
+          var state = "loading";
+          if (document.querySelectorAll("span.replies").length > 0) {
```

Comment:
I don't love this approach - I think it would be better if the rails app rendered the state into the DOM in data-* attributes somewhere and the javascript read those for the state. Otherwise it's tied to the specific elements in the DOM.

Reply:
Is there a part of the DOM that can be read inside and outside of the iframe ? If there is then I'm fine with reading that info. Still sending messages seems nice because otherwise the script from the outside needs to regularly check the DOM. In case of a message it can just sit and wait until a message arrives. But I'm not really deep into messaging in javascript. Maybe there's a possibility to trigger a message on change of a data attribute ?

Reply:
@eviltrout what I just wrote was about the communication between javascript inside the iframe and javascript outside of the iframe. Rereading your comment I now assume that I got you wrong. 
There is
I) Rails
II) javascript inside the iframe
--- iframe border ---
III) javascript outside the iframe

I think it would be very nice if I) rails modified the DOM (inside the iframe) setting a clearly defined state flag.
Then the II) javascript inside the iframe could send a message through the iframes border including the state flag.
That message could then be read outside the iframe III).

This pull request is a hack because it doesn't respect the rails level. I very much appreciate a proper approach on the rails level.

Reply:
What I was suggesting is mostly the same as your PR. Instead of counting elements, just look for a DOM element. Rails generates the contents of the iframe, so it can drop a data-* element in there when it renders the contents. The javascript can look for that element, and `postUp`.